tensorflow/python/data/experimental/ops/get_single_element.py

```diff
@@ -20,31 +20,119 @@
 from tensorflow.python.data.ops import dataset_ops
 from tensorflow.python.data.util import structure
 from tensorflow.python.ops import gen_dataset_ops
+from tensorflow.python.util import deprecation
 from tensorflow.python.util.tf_export import tf_export
 
 
-@tf_export("data.experimental.get_single_element")
+@tf_export("data.experimental.get_single_element", "data.get_single_element")
+@deprecation.deprecated_endpoints("data.experimental.get_single_element")
```

Comment:
Similar to my comments on `bucket_by_sequence_length`, I would prefer exposing this in the core API as a method on the `Dataset` class. Thus the usage becomes `dataset.get_single_element()` as opposed to `tf.data.get_single_element(dataset)`. You can see how `reduce` is implemented in dataset_ops.py as it is similar to `get_single_element` in the sense that it provide a way to consume the dataset element(s) and the usage (and implementation) of `get_single_element` should ideally be consistent with `reduce`.

Reply:
@jsimsa sure will take a look.